src/Util/NumberOfCores.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Php\Pie\Util;
+
+use Symfony\Component\Process\Process;
+
+use function is_numeric;
+use function trim;
+
+final class NumberOfCores
+{
+    /** @psalm-suppress UnusedConstructor */
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param list<string> $params
+     *
+     * @return positive-int
+     */
+    private static function tryCommand(string $command, array $params): int|null
+    {
+        $whichProcess = new Process(['which', $command]);
+        if ($whichProcess->run() === 0) {
+            $commandPath = trim($whichProcess->getOutput());
+
+            $commandProcess = new Process([$commandPath, ...$params]);
+            if ($commandProcess->run() === 0) {
+                $commandResult = trim($commandProcess->getOutput());
+                if (is_numeric($commandResult)) {
+                    $commandNumericResult = (int) $commandResult;
+
+                    if ($commandNumericResult > 0) {
+                        return $commandNumericResult;
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Try a few different ways of determining number of CPUs. If we can't guess
+     * how many CPUs someone has, then default to 1 for safety.
+     *
+     * @return positive-int
+     */
+    public static function determine(): int
```

Comment:
I recommend https://github.com/theofidry/cpu-core-counter, its the code used in most tools which have parallelisation

Reply:
Some great packages already used it : see https://packagist.org/packages/fidry/cpu-core-counter/dependents?order_by=downloads
And I hope (me too) to adapt some of my packages to add a parallelization feature